package/Shaders/Water.hlsl

```diff
@@ -633,16 +732,17 @@ PS_OUTPUT main(PS_INPUT input)
 #			else
 	distanceMul = 0;
 
-	depth = GetScreenDepthWater(
-		DynamicResolutionParams1.xy * (DynamicResolutionParams2.xy * input.HPosition.xy));
+	float2 screenPosition = DynamicResolutionParams1.xy * (DynamicResolutionParams2.xy * input.HPosition.xy);
+
+	depth = GetScreenDepthWater(screenPosition);
 	float2 depthOffset =
 		DynamicResolutionParams2.xy * input.HPosition.xy * VPOSOffset.xy + VPOSOffset.zw;
 	float depthMul = length(float3((depthOffset * 2 - 1) * depth / ProjData.xy, depth));
 
 	float3 depthAdjustedViewDirection = -viewDirection * depthMul;
-	float viewSurfaceAngle = dot(depthAdjustedViewDirection, ReflectPlane.xyz);
+	float viewSurfaceAngle = dot(depthAdjustedViewDirection, ReflectPlane[eyeIndex].xyz);
```

Comment:
This and the next line appear to be where this will break. If these are set to a static float value, then the water depth will be wrong, but it will not swim with the headset.

Reply:
ReflectPlane[eyeIndex] is fine. Both 0 and 1 have the exact same value.

Reply:
Fixed, `depthMul` needs to be calculated differently in VR.